src/libfetchers/attrs.hh

```diff
@@ -12,6 +12,9 @@
 
 namespace nix::fetchers {
 
+/**
+ * A primitive value that can be used in a fetcher attribute.
+ */
 typedef std::variant<std::string, uint64_t, Explicit<bool>> Attr;
```

Comment:
In some later PR we should probably do a wrapper struct (per https://github.com/NixOS/nix/issues/7479#issuecomment-1490977948) so the functions below can be methods.